src/app/beer_garden/topic.py

```diff
@@ -184,14 +184,26 @@ def subscriber_match(
     return match
 
 
-def prune_topics(garden):
+def topic_has_system_subscribers(topic: Topic, system: System):
+    for subscriber in topic.subscribers:
+        if subscriber.system == system.name:
```

Comment:
This should check `system.name`, `system.namespace`, and `system.version`. Because if you have two versions of a System running, it could delete the dynamic topics from the other one.